stubs/python-decouple/decouple.pyi

```diff
@@ -0,0 +1,168 @@
+    def __contains__(self, key: str) -> bool: ...
+    def __getitem__(self, key: str) -> str: ...
+
+class RepositoryIni(RepositoryEmpty):
+    SECTION: str
+    parser: ConfigParser
+    def __init__(self, source: str, encoding: str = ...) -> None: ...
+    def __contains__(self, key: str) -> bool: ...
+    def __getitem__(self, key: str) -> str: ...
+
+class RepositoryEnv(RepositoryEmpty):
+    data: dict[str, str]
+    def __init__(self, source: str, encoding: str = ...) -> None: ...
+    def __contains__(self, key: str) -> bool: ...
+    def __getitem__(self, key: str) -> str: ...
+
+class RepositorySecret(RepositoryEmpty):
+    data: dict[str, str]
+    def __init__(self, source: str = ...) -> None: ...
+    def __contains__(self, key: str) -> bool: ...
+    def __getitem__(self, key: str) -> str: ...
+
+class AutoConfig:
+    SUPPORTED: OrderedDict[str, RepositoryEmpty]
+    encoding: str
+    search_path: str | None
+    config: Config | None
+    def __init__(self, search_path: str | None = None) -> None: ...
+    @overload
+    def __call__(self, option: str, default: str = ...) -> str: ...
+    @overload
+    def __call__(self, option: str, default: _T) -> str | _T: ...
+    @overload
+    def __call__(self, option: str, *, cast: Callable[[str], _T]) -> _T: ...
+    @overload
+    def __call__(self, option: str, default: str | _T, cast: Callable[[str], _T]) -> _T: ...
+
+config: AutoConfig
+
+class Csv(Generic[_T, _TCsv]):
+    cast: Callable[[str], _T]
+    delimiter: str
+    strip: str
+    post_process: Callable[[Iterable[Any]], _TCsv]
+    def __call__(self, value: str) -> _TCsv: ...
+    @overload
+    def __init__(
+        self: Csv[str, list[str]],
+        cast: Callable[[str], str] = ...,
+        delimiter: str = ...,
+        strip: str = ...,
+        post_process: Callable[[Iterable[Any]], list[str]] = ...,
+    ) -> None: ...
+    @overload
+    def __init__(
+        self: Csv[_T, list[_T]],
+        cast: Callable[[str], _T],
+        delimiter: str = ...,
+        strip: str = ...,
+        post_process: Callable[[Iterable[Any]], list[_T]] = ...,
+    ) -> None: ...
+    @overload
+    def __init__(
+        self: Csv[str, _TCsv],
+        cast: Callable[[str], str] = ...,
+        delimiter: str = ...,
+        strip: str = ...,
+        *,
+        post_process: Callable[[Iterable[Any]], _TCsv],
+    ) -> None: ...
+    @overload
+    def __init__(
+        self: Csv[_T, _TCsv],
+        cast: Callable[[str], _T],
+        delimiter: str = ",",
+        strip: str = ...,
+        *,
+        post_process: Callable[[Iterable[Any]], _TCsv],
+    ) -> None: ...
+    @overload
+    def __init__(
+        self: Csv[_T, _TCsv],
+        cast: Callable[[str], _T],
+        delimiter: str,
+        strip: str,
+        post_process: Callable[[Iterable[Any]], _TCsv],
+    ) -> None: ...
+
+class Choices(Generic[_T]):
+    flat: Sequence[_T]
+    cast: Callable[[str], _T]
+    choices: Sequence[_T]
```

Comment:
Shouldn't this be `Sequence[tuple[_T, object]]`?

Also in the overloads below I think the second tuple member in `choices` should be `object`, as the implementation ignores it.

Reply:
Indeed, I had already suggested the same change for `choices` but I guess it fell victim to manual copy pasta.

While I do agree that the second member in the tuple is ignored and could be anything, I don't think it's a good change.
It expects that when `cast` is used on the second element in the tuple, for it to return the first element, so while the implementation is bogus and backwards and doesn't verify that, the stubs should probably reflect the intent and the intent was for the second element to be the string representation of the choice.